File: leetcode/problems/c/single-number.c
// by @codeAbinash
// Time : O(n)
// Space : O(1)

int singleNumber(int* nums, int numsSize) {
   int singleNumber = nums[0];
   for (int i = 1; i < numsSize; i++)
      singleNumber ^= nums[i];
   return singleNumber;
}

File: leetcode/problems/ts/single-number.ts
// by @codeAbinash
// Time : O(n)
// Space : O(1)

function singleNumber(nums: number[]): number {
   let result = nums[0]
   let len = nums.length;
   for (let i = 1; i < len; i++)
      result ^= nums[i]
   return result
}